src/Disks/IDisk.cpp

```diff
@@ -251,6 +251,7 @@ catch (Exception & e)
 void IDisk::applyNewSettings(const Poco::Util::AbstractConfiguration & config, ContextPtr /*context*/, const String & config_prefix, const DisksMap & /*map*/)
```

Comment:
Looks like I moved it to `IDisk` when transaction's `removeSharedRecursive` method accepted batch size as a parameter in case of `DiskEncrypted` over object storage disk ([here](https://github.com/ClickHouse/ClickHouse/blob/558bec1c94eced66ad44920957a118c8b26a4fa0/src/Disks/DiskEncrypted.h#L178)).
In current implementation I can move it to `DiskObjectStorage`, but if we pass batch size as an argument to transaction function, as proposed below, we should leave it here or maybe try to cast `DiskEncrypted::delegate` disk to `DiskObjectStorage` and obtain batch size this way.